klient/decoder/decoder_test.go

```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package decoder
+package decoder_test
```

Comment:
Had to modify some of these to avoid dependency import cycles. These are standard go convention anyway. Switched only the selective bits to this format.

Reply:
While reviewing yesterday, I wanted to bring up that issue.  (Even if not in this PR) We should take a look at reorg the package layout.   

For instance, pkg/envfunc should not depend support/kind (but I understand that's how we did at first). 
All pkg packages should be depended on by other external packages (not the other way around).  Otherwise we will keep getting into these tricky circular dependencies.

Reply:
@vladimirvivien  +1. I totally agree. I believe we need to reorg the packages a bit to cleanup some of these. I remember we had this discussion as part of a different Git Issue but did not take that forward then. Let us open a Git Tracker issue and sort that out right away. I also want to start using the `_test` convention in the packages for the Tests if that is all right with everyone, Makes it more cleaner in terms of organizing and avoid a lot of dependency issues too.